### PR TITLE
feat: remove dependence on Bluebird and delay #67

### LINKED ISSUE #67: Remove Bluebird
Just from a brief glance, it looks like only `delay` is being used from Bluebird.  It really is a no-starter for us to include libs with Bluebird since it seriously hinders the async guarantee in the server env.

Is this something I could help with?

PS: so excited to see a rewrite of Bull!  Thanks for all the hard work

### PR DESCRIPTION
**Pull request by jbr**
This approach uses a simple setTimeout-based delay in the only production
usage (src/classes/queue-events.ts), defined in src/util.ts, and since
there was already a devDependency on a library called
[delay](https://www.npmjs.com/package/delay), this commit uses that in
test contexts. It should be easy to replace that dep with a
`(ms:number) => new Promise(r => setTimeout(r, ms))` as well.

Closes #67

### CODE REVIEW

File: src/test/fixtures/delay.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = function delay(ms) {
```
Comment (manast) [inline src/test/fixtures/delay.js]: why cant we use the delay implementation in utils?

Comment (jbr) [reply src/test/fixtures/delay.js]: That would be requiring ts from js, which seemed like a substantial change to the fixtures

Comment (manast) [reply src/test/fixtures/delay.js]: ah ok, got it.

**Pull request merged**